proposals/NNNN-Removing-URL-Param-Max-Length.md

```diff
@@ -0,0 +1,84 @@
+
+## Mobile API
+
+We should remove the max length attribute of the following URL for OnSystemRequest. 
+
+```xml
+   <function name="OnSystemRequest" functionID="OnSystemRequestID" messagetype="notification" >
+        <description>
+            An asynchronous request from the system for specific data from the device or the cloud or response to a request from the device or cloud
+            Binary data can be included in hybrid part of message for some requests (such as Authentication request responses)
+        </description>
+        <param name="requestType" type="RequestType" mandatory="true">
+            <description>The type of system request.</description>
+        </param>
+        <param name="url" type="String" mandatory="false">
+            <description>
+                Optional URL for HTTP requests.
+                If blank, the binary data shall be forwarded to the app.
+                If not blank, the binary data shall be forwarded to the url with a provided timeout in seconds.
+            </description>
+        </param>
+        ...
+```   
+
+## HMI API
+
+We should remove the max length attribute of the following URL for OnSystemRequest in HMI API. 
+
+```xml
+   <function name="OnSystemRequest" messagetype="notification" >
+      <description>
+        An asynchronous request from the system for specific data from the device or the cloud or response to a request from the device or cloud
+        Binary data can be included in hybrid part of message for some requests (such as Authentication request responses)
+      </description>
+      <param name="requestType" type="Common.RequestType" mandatory="true">
+        <description>The type of system request.</description>
+      </param>
+      <param name="url" type="String" minlength="1" mandatory="false">
+        <description>Optional array of URL(s) for HTTP requests.</description>
+      </param>
+      ...
+```   
+
+The URL parameters in StartStream and StartAudioStream in HMI API should also have max length removed.
+
+```xml
+  <function name="StartStream" messagetype="request">
+    <description>Request from SmartDeviceLinkCore to HMI to start playing video streaming.</description>
+    <param name="url" type="String" minlength="21" mandatory="true">
+      <description>URL that HMI start playing.</description>
+    </param>
+    ...
+  <function name="StartAudioStream" messagetype="request">
+   <description>Request from SmartDeviceLinkCore to HMI to start playing audio streaming.</description>
+   <param name="url" type="String" minlength="21" mandatory="true">
+     <description>URL that HMI start playing.</description>
+   </param>
+   ...
+```
+
+# Potential downsides
+
+Increasing the size of the URL may provide a minimal performance impact since the system needs to process a larger string for the URL parameter. The increased string length may also be considered a breaking change.
+
+# Alternatives considered
```

Comment:
This should be an H2 header (##)

Reply:
Changed to H2 in 5fb96551bc4d6fb18defff370ab83a763dd7eb62.